tsl/src/continuous_aggs/create.c

```diff
@@ -2410,6 +2571,14 @@ cagg_create(const CreateTableAsStmt *create_stmt, ViewStmt *stmt, Query *panquer
 	bool finalized = DatumGetBool(with_clause_options[ContinuousViewOptionFinalized].parsed);
 
 	finalqinfo.finalized = finalized;
+	if (list_length(panquery->jointree->fromlist) > 1 && !materialized_only)
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("for continuous aggregates with joins,"
+						"real-time aggregates is not supported"),
+				 errhint("set materialized_only to true")));
```

Comment:
Sorry, but why we're preventing users to use JOIN in real-time continuous aggregates? At least we should add test cases for that.

Reply:
For this PR we have decided to keep it limited to caggs without real time aggregates. For real time aggregates we have to go through the additional changes for union query, etc. a follow up pr would be better way to do this. But yes I will add the test case for this.